crates/services/executor/src/executor.rs

```diff
@@ -548,9 +554,22 @@ where
         let block_height = *block.header.height();
 
         if self.relayer.enabled() {
-            self.process_da(block_st_transaction, &block.header, execution_data)?;
+            self.process_da(&mut block_st_transaction, &block.header, execution_data)?;
         }
 
+        // The block level storage transaction that also contains data from the relayer.
+        // Starting this point modifications from each thread should be independent
+        // and shouldn't touch the same data.
+        let mut block_with_relayer_data_transaction = block_st_transaction.write_transaction()
+                // Enforces independent changes from each thread.
+                .with_policy(ConflictPolicy::Fail);
+
+        // We execute transactions in a single thread right now, but later,
```

Comment:
Is there an issue for tracking this?

Reply:
https://github.com/FuelLabs/fuel-core/issues/1088 and https://github.com/FuelLabs/fuel-core/issues/1189